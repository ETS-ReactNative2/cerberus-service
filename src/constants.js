export const LONG_DATE_FORMAT = 'D MMM YYYY [at] HH:mm';
export const SHORT_DATE_FORMAT = 'DD/MM/YYYY';
export const FORM_NAME_TARGET_INFORMATION_SHEET = 'targetInformationSheet';
export const TASK_STATUS_NEW = 'new';
export const TASK_STATUS_IN_PROGRESS = 'inProgress';
export const TASK_STATUS_TARGET_ISSUED = 'issued';
export const TASK_STATUS_COMPLETED = 'complete';
export const TARGETER_GROUP = '/bf-intel-targeters';
export const RORO_TOURIST = 'RORO_Tourist';
export const RORO_UNACCOMPANIED_FREIGHT = 'RORO_Unaccompanied_Freight';
export const RORO_ACCOMPANIED_FREIGHT = 'RORO_Accompanied_Freight';
export const RORO_TOURIST_CAR_ICON = 'c-icon-car';
export const RORO_TOURIST_INDIVIDUAL_ICON = 'c-icon-person';
export const RORO_TOURIST_GROUP_ICON = 'c-icon-group';
export const RORO_UNACCOMPANIED_ICON = 'c-icon-trailer';
export const RORO_ACCOMPANIED_ICON = 'c-icon-hgv';
export const RORO_VAN_ICON = 'c-icon-van';
export const RORO_NO_ICON = '';
