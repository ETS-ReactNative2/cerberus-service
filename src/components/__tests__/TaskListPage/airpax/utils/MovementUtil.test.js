import renderer from 'react-test-renderer';

import MovementUtil from '../../../../TaskListPage/airpax/utils/movementUtil';
import { UNKNOWN_TEXT } from '../../../../../constants';

describe('MovementUtil', () => {
  const targetTaskMin = {
    movement: {
      id: 'AIRPAXTSV:CMID=9c19fe74233c057f25e5ad333672c3f9/2b4a6b5b08ea434880562d6836b1111',
      status: 'PRE_ARRIVAL',
      mode: 'AIR_PASSENGER',
      description: 'individual',
      journey: {
        id: 'BA103',
        arrival: {
          country: null,
          location: 'LHR',
          time: null,
        },
        departure: {
          country: null,
          location: 'FRA',
          time: '2020-08-07T17:15:00Z',
        },
        route: [
          'FRA',
          'LHR',
        ],
        itinerary: [
          {
            id: 'BA103',
            arrival: {
              country: null,
              location: 'LHR',
              time: null,
            },
            departure: {
              country: null,
              location: 'FRA',
              time: '2020-08-07T17:15:00Z',
            },
          },
        ],
      },
      flight: {
        departureStatus: 'DC',
        number: 'BA103',
        operator: 'BA',
        seatNumber: null,
      },
    },
  };

  it('should return the route if present', () => {
    const output = MovementUtil.movementRoute(targetTaskMin.movement.journey);
    expect(output).not.toBeUndefined();
    expect(output).not.toBeNull();
    expect(output).toEqual(targetTaskMin.movement.journey.route);
  });

  it('should return null if formatted seat number is not present', () => {
    const output = MovementUtil.seatNumber(targetTaskMin.movement.flight);
    expect(output).toEqual(`seat ${UNKNOWN_TEXT}`);
  });

  it('should return null if formatted seat number is present', () => {
    targetTaskMin.movement.flight.seatNumber = '15C';

    const output = MovementUtil.seatNumber(targetTaskMin.movement.flight);
    expect(output).toEqual('seat 15C');
  });

  it('should return a flight object if present', () => {
    const output = MovementUtil.movementFlight(targetTaskMin);
    expect(output).not.toBeUndefined();
    expect(output).not.toBeNull();
    expect(output).toEqual(targetTaskMin.movement.flight);
  });

  it('should return a journey object if present', () => {
    const output = MovementUtil.movementFlight(targetTaskMin);
    expect(output).not.toBeUndefined();
    expect(output).not.toBeNull();
    expect(output).toEqual(targetTaskMin.movement.flight);
  });

  it('should return a departure time if present', () => {
    const output = MovementUtil.departureTime(targetTaskMin.movement.journey);
    expect(output).toEqual(targetTaskMin.movement.journey.departure.time);
  });

  it('should return arrival time if present', () => {
    const output = MovementUtil.arrivalTime(targetTaskMin.movement.journey);
    expect(output).toEqual(targetTaskMin.movement.journey.arrival.time);
  });

  it('should return a formatted departure date and time if present', () => {
    const output = MovementUtil.formatDepartureTime(targetTaskMin.movement.journey);
    expect(output).toEqual('7 Aug 2020 at 18:15');
  });

  it('should return a formatted arrival date and time if present', () => {
    const output = MovementUtil.formatArrivalTime(targetTaskMin.movement.journey);
    expect(output).toEqual(UNKNOWN_TEXT);
  });

  it('should return a departure location if present', () => {
    const output = MovementUtil.departureLoc(targetTaskMin.movement.journey);
    expect(output).toEqual(targetTaskMin.movement.journey.departure.location);
  });

  it('should return an arrival location if present', () => {
    const output = MovementUtil.arrivalLoc(targetTaskMin.movement.journey);
    expect(output).toEqual(targetTaskMin.movement.journey.arrival.location);
  });

  it('should return a flight number if present', () => {
    const output = MovementUtil.flightNumber(targetTaskMin.movement.flight);
    expect(output).toEqual(targetTaskMin.movement.flight.number);
  });

  it('should return the airline operator if present', () => {
    const output = MovementUtil.airlineOperator(targetTaskMin.movement.flight);
    expect(output).toEqual(targetTaskMin.movement.flight.operator);
  });

  it('should render the departure status if present', () => {
    const tree = renderer.create(MovementUtil.status(targetTaskMin)).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should return the movement type if present', () => {
    const output = MovementUtil.movementType(targetTaskMin);
    expect(output).toEqual('Passenger');
  });

  it('should return the movement description text', () => {
    const output = MovementUtil.description(targetTaskMin);
    expect(output).toEqual('Single passenger');
  });

  it('should return the movement description text for multiple persons', () => {
    targetTaskMin.movement.description = 'group';

    targetTaskMin.movement.person = {
      name: {
        first: 'Isaiah',
        last: 'Ford',
        full: 'Isaiah Ford',
      },
      role: 'PASSENGER',
      dateOfBirth: '1966-05-13T00:00:00Z',
      gender: 'M',
      nationality: 'GBR',
      document: null,
    };

    targetTaskMin.movement.otherPersons = [
      {
        name: {
          first: 'Isaiah',
          last: 'Ford',
          full: 'Isaiah Ford',
        },
        role: 'PASSENGER',
        dateOfBirth: '1966-05-13T00:00:00Z',
        gender: 'M',
        nationality: 'GBR',
        document: null,
      },
    ];

    const output = MovementUtil.description(targetTaskMin);
    expect(output).toEqual('In group of 2');
  });

  it('should convert airline code to airline name', () => {
    const GIVEN_BA = MovementUtil.airlineName('BA');
    const GIVEN_AF = MovementUtil.airlineName('AF');
    const GIVEN_QF = MovementUtil.airlineName('QF');

    expect(GIVEN_BA).toEqual('British Airways');
    expect(GIVEN_AF).toEqual('Air France');
    expect(GIVEN_QF).toEqual('Qantas');
  });
});
