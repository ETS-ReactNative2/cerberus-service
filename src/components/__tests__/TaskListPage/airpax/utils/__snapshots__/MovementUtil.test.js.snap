// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovementUtil should render the departure status if present 1`] = `
<span
  className="govuk-body govuk-tag govuk-tag--departureConfirmed"
>
  DC
</span>
`;

exports[`MovementUtil should return the departure status if present 1`] = `
<span
  className="govuk-body govuk-tag govuk-tag--departureConfirmed"
>
  DC
</span>
`;
